lib/usage-tracking/class-usage-tracking.php

```diff
@@ -0,0 +1,454 @@
+	 * data to be logged periodically to Tracks.
+	 *
+	 * @param callable the callback returning the usage data to be logged.
+	 **/
+	public function set_callback( $callback ) {
+		$this->callback = $callback;
+	}
+
+	/**
+	 * Send an event to Tracks if tracking is enabled.
+	 *
+	 * @param string $event The event name.
+	 * @param array $properties Event Properties.
+	 * @param null|int $event_timestamp When the event occurred.
+	 * @param bool $force if true, send the event even if tracking is disabled.
+	 *
+	 * @return null|WP_Error
+	 **/
+	public function send_event( $event, $properties = array(), $event_timestamp = null, $force = false ) {
+
+		// Only continue if tracking is enabled, or send_event is forced
+		if ( ! $this->is_tracking_enabled() && ! $force ) {
+			return false;
+		}
+
+		$pixel = 'http://pixel.wp.com/t.gif';
+		$event_prefix = self::PREFIX . '_';
+		$event_name = $event_prefix . str_replace( $event_prefix, '', $event );
+		$user = wp_get_current_user();
+
+		if ( null === $event_timestamp ) {
+			$event_timestamp = time();
+		}
+
+		$properties['admin_email'] = get_option( 'admin_email' );
+		$properties['_ut'] = self::PREFIX . ':site_url';
+		// Use site URL as the userid to enable usage tracking at the site level.
+		// Note that we would likely want to use site URL + user ID for userid if we were
+		// to ever add event tracking at the user level.
+		$properties['_ui'] = site_url();
+		$properties['_ul'] = $user->user_login;
+		$properties['_en'] = $event_name;
+		$properties['_ts'] = $event_timestamp . '000';
+		$properties['_rt'] = round( microtime( true ) * 1000 );  // log time
+		$p = array();
+
+		foreach( $properties as $key => $value ) {
+			$p[]= urlencode( $key ) . '=' . urlencode( $value );
+		}
+
+		$pixel .= '?' . implode( '&', $p ) . '&_=_'; // EOF marker
+		$response = wp_remote_get( $pixel, array(
+			'blocking'    => true,
+			'timeout'     => 1,
+			'redirection' => 2,
+			'httpversion' => '1.1',
+			'user-agent'  => self::PREFIX . '_usage_tracking',
+		) );
+
+		if ( is_wp_error( $response ) ) {
+			return $response;
+		}
+
+		$code = isset( $response['response']['code'] ) ? $response['response']['code'] : 0;
+
+		if ( $code !== 200 ) {
+			return new WP_Error( 'request_failed', 'HTTP Request failed', $code );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Set up a regular cron job to send usage data. The job will only send
+	 * the data if tracking is enabled, so it is safe to call this function,
+	 * and schedule the job, before the user opts into tracking.
+	 **/
+	public function schedule_tracking_task() {
+		if ( ! wp_next_scheduled( $this->job_name ) ) {
+			wp_schedule_event( time(), self::PREFIX . '_usage_tracking_two_weeks', $this->job_name );
+		}
+	}
+
+	/**
+	 * Unschedule the job scheduled by schedule_tracking_task if any is
+	 * scheduled. This should be called on plugin deactivation.
+	 **/
+	public function unschedule_tracking_task() {
```

Comment:
Do we ever hook this into plugin deactivation

Reply:
Yes - https://github.com/Automattic/sensei/blob/master/includes/class-sensei.php#L529

(Hey look at that. The method name is wrong there. Thanks for asking 😄 )